crates/red_knot_python_semantic/src/types/generics.rs

```diff
@@ -130,6 +148,15 @@ impl<'db> GenericContext<'db> {
         self.specialize(db, types.into())
     }
 
+    pub(crate) fn is_subset_of(self, db: &'db dyn Db, other: GenericContext<'db>) -> bool {
+        for variable in self.variables(db) {
+            if !other.variables(db).iter().contains(variable) {
+                return false;
+            }
+        }
```

Comment:
This is `O(n^2)`. Can we short circuit, e.g. compare the lengths first?

It might also be worth to cache `other.variables(db)` and move it outside the loop

Reply:
Yeah I hadn't liked the quadratic nature of this ever since I first wrote it.  Turns out we've been using `FxOrderSet` to build up the variable list in most places already, so I've updated this to just store that set in `GenericContext`.  Now we can use its [`is_subset`](https://docs.rs/ordermap/latest/ordermap/set/struct.OrderSet.html#method.is_subset) method to do the work.